chaosk8s/pod/actions.py

```diff
@@ -46,21 +49,120 @@ def terminate_pods(label_selector: str = None, name_pattern: str = None,
     be used as the grace period (in seconds) to terminate the pods.
     Otherwise, the default pod's grace period will be used.
     """
+
+    api = create_k8s_api_client(secrets)
+    v1 = client.CoreV1Api(api)
+
+    pods = _select_pods(v1, label_selector, name_pattern,
+                        all, rand, mode, qty, ns, order)
+
+    body = client.V1DeleteOptions()
+    if grace_period >= 0:
+        body = client.V1DeleteOptions(grace_period_seconds=grace_period)
+
+    for p in pods:
+        v1.delete_namespaced_pod(p.metadata.name, ns, body=body)
+
+
+def exec_in_pods(label_selector: str = None, name_pattern: str = None,
```

Comment:
It would be very useful if the function could return the output of the call. So this could be used as a probe in the steady state hypothesis to verify stuff from within the container.

Maybe return, like a process:

```
return {"status": CODE, "stdout": ..., "stderr": ...}
```

This would map the process type return value.

Reply:
Hi, well am not sure how to handle this, other then logging the stderr and stdout for each call.
The call is inside a loop (a call per matching pod). It does not make sense return just the last call result.
Two solutions here:
solutions 1)  fusion all results in a dictionary indexed by podname and return the dictionary
solution 2) stop the loop when one of the call fails and return that results only to indicate the error. 
May be you have a better idea ?

Reply:
I think 1/ is better. No reason to fail as soon as one calls fails I'd say. However, I wouldn't return a dict keyed by podname because they can't be guessed and would be difficult to use in a tolerance probe.

I'd return then a list of objects:

```
[
  {
     "pod_name": "...",
     "exit_code": "...",
     "cmd": "...",
     "stdout", "...",
     "stderr": "..."
  }
]
```

That way, we could apply a jsonpath against this more easily I believe.

wdyt?

Reply:
Adopted! the function returns now a list of dicts as per your example